### PR TITLE
docs: dt-531 component combinator

### PR DESCRIPTION
**Pull request by jonathan-paugh**
## Description
Implement the [dialtone-combinator](https://github.com/dialpad/dialtone-combinator) package in the dialtone documentation website. 

<br />

To include the combinator in markdown simply use the tag like you would in a vue template, passing in the target component to use:

```vue
<component-combinator component-name="DtButton" />
```

<br />

Currently displayed below the initial preview on the component pages, however this can be changed easily.

image

<br />

Uses variants from the `variantBank` export in the dialtone-combinator package, this is a good place to keep them as they will be shared between whatever uses the combinator in the future. However if needed these can be customized by passing custom data into the combinator `variants` prop.

image

Clear variant:
image

<br />

The `showBlueprints` prop can also be used to show a group of variants in a container below the combinator, leveraging the combinators 'blueprint' mode.

```vue
<component-combinator component-name="DtButton" show-blueprints />
```

image

image

This is currently not used anywhere however could be very useful for easily conveying the code required to create different variants, which is a problem in storybook right now.

## Pull Request Checklist

 - [x] Ask the contributors if a similar effort is already in process or has been solved.
 - [x] Review the [contribution guidelines](https://github.com/dialpad/dialtone/blob/staging/.github/CONTRIBUTING.md).
 - [x] Use `staging` as your pull request's base branch. (All PRs using `production` as its base branch will be declined).
 - [x] Ensure all `gulp` scripts successfully compile.
 - [x] Update, remove, or extend all affected documentation.
 - [x] Ensure no private Dialpad links or info are in the code or pull request description (Dialtone is a public repo!).

### CODE REVIEW

File: docs/components/tooltip.md [hunk 1]
```diff
@@ -12,6 +12,8 @@ storybook_url: https://vue.dialpad.design/?path=/story/components-tooltip--defau
   </button>
 </code-well-header>
 
+<component-combinator component-name="DtTooltip" />
```
Comment (josedialpad) [inline docs/components/tooltip.md]: It warns with "Unsupported component"?

File: docs/components/avatar.md [hunk 2]
```diff
@@ -12,6 +12,8 @@ prev:
   </div>
 </code-well-header>
 
+<component-combinator component-name="DtAvatar" />
```
Comment (josedialpad) [inline docs/components/avatar.md]: Missing the Banner, Modal, Popover, Tabs, skeleton, or are these not supported?

**Review by josedialpad — COMMENTED** [review]
really great work on the combinator project!

jose

**Comment by jonathan-paugh** [comment]
> really great work on the combinator project!
> 
> jose

Thanks! I'm honoured to get the legendary "Jose approves" gif 🤣

File: docs/components/tooltip.md (hunk 1 above)
Comment (jonathan-paugh) [reply docs/components/tooltip.md]: Yeah, I guess this one should be removed for now, it mostly works it just seems like any of the components the open/close break and stay opened after bit for some reason.

File: docs/components/avatar.md (hunk 2 above)
Comment (jonathan-paugh) [reply docs/components/avatar.md]: Yeah the unsupported ones are not included currently.

**Comment by jonathan-paugh** [comment]
It finally works on the deploy preview!

**Comment by braddialpad** [comment]
Sweet! looks like everything is working. We should probably hold off on actually merging this at the moment as there are probably some design changes we need to make before publishing it so it fits in visually with the rest of the site. But by doing this we've still figured out everything we had to in order to get this published an implemented in the doc site. We can just keep this PR open for the time being.

**Comment by braddialpad** [comment]
Just so we don't have to have this PR sitting here forever and getting way out of date with staging, I've commented out the usage of combinator on the pages themselves and I'm going to merge it in. Should have no visual change to the current doc site.

**Pull request merged**